OneSignalSDK/onesignal/src/main/java/com/onesignal/OSInAppMessageController.java

```diff
@@ -409,6 +420,53 @@ else if (action.getUrlTarget() == OSInAppMessageAction.OSInAppMessageActionUrlTy
         }
     }
 
+    private void fireRESTCallForPageChange(@NonNull final OSInAppMessage message, @NonNull final OSInAppMessagePage page) {
+        final String variantId = variantIdForMessage(message);
+        if (variantId == null)
+            return;
+
+        final String pageId = page.getPageId();
+
+
+        // Never send multiple page impressions for the same message UUID unless that page change is from an IAM with redisplay
+        if (message.getViewedPageIds().contains(pageId))
+            return;
+
+        message.addPageId(page.getPageId());
+
+        try {
+            JSONObject json = new JSONObject() {{
+                put("app_id", OneSignal.appId);
+                put("player_id", OneSignal.getUserId());
+                put("variant_id", variantId);
+                put("device_type", new OSUtils().getDeviceType());
+                put("page_id", pageId);
+            }};
+
+            OneSignalRestClient.post("in_app_messages/" + message.messageId + "/pageImpression", json, new ResponseHandler() {
+                @Override
+                void onSuccess(String response) {
+                    printHttpSuccessForInAppMessageRequest("page impression", response);
+                    /*OneSignalPrefs.saveStringSet(
```

Comment:
Looks like this is a TODO yet?

Reply:
Yes this was my question at the end of the PR do we need to save the ids to the prefs?

Reply:
The only scenario that I think needs this.
1 - Opening app 
2 - Display IAM
3 - Impression sent
4 - Close application (swipe away) without dismiss IAM
5 - Open app
6 - IAM should display again
7 - Impression shouldn't be sent again

Reply:
Cool confirmed that this is an issue so I will store them in the prefs